pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -142,6 +154,69 @@ func (e *cloudWatchExecutor) getRGTAClient(region string) (resourcegroupstagging
 	return e.rgtaClient, nil
 }
 
+func (e *cloudWatchExecutor) getServiceQuotasClient(region string) (servicequotasiface.ServiceQuotasAPI, error) {
+	sess, err := e.newSession(region)
+	if err != nil {
+		return nil, err
+	}
+
+	return newQuotasClient(sess), nil
+}
+
+func (e *cloudWatchExecutor) getQueue(region string) (chan bool, error) {
+	e.queueLock.Lock()
+	defer e.queueLock.Unlock()
+
+	if queue, ok := e.queuesByRegion[region]; ok {
+		return queue, nil
+	}
+
+	concurrentQueriesQuota, err := e.fetchConcurrentQueriesQuota(region)
+	if err != nil {
+		plog.Info("Could not fetch quota")
+	}
+
+	queueChannel := make(chan bool, concurrentQueriesQuota)
+	e.queuesByRegion[region] = queueChannel
+
+	return queueChannel, nil
+}
+
+func (e *cloudWatchExecutor) fetchConcurrentQueriesQuota(region string) (int, error) {
+	client, err := e.getServiceQuotasClient(region)
+	if err != nil {
+		return defaultConcurrentQueries, err
+	}
+
+	concurrentQueriesQuota, err := client.GetServiceQuota(&servicequotas.GetServiceQuotaInput{
+		ServiceCode: aws.String("logs"),
+		QuotaCode:   aws.String("L-32C48FBB"),
+	})
+
+	if err != nil {
+		return defaultConcurrentQueries, err
+	}
```

Comment:
Error checks should be co-located with the error's origin, for readability:
```suggestion
	concurrentQueriesQuota, err := client.GetServiceQuota(&servicequotas.GetServiceQuotaInput{
		ServiceCode: aws.String("logs"),
		QuotaCode:   aws.String("L-32C48FBB"),
	})
	if err != nil {
		return defaultConcurrentQueries, err
	}
```